sycl/doc/extensions/SPIRV/SPV_INTEL_float_controls2.asciidoc

```diff
@@ -0,0 +1,265 @@
+If an instruction is decorated with *FPRoundingMode* or defines a rounding mode in its description, that rounding mode is applied and *RoundingModeRTNINTEL* is ignored.
+If an instruction is not decorated with *FPRoundingMode* and does not define a rounding mode in its description and is inside of a function decorated
+with *FunctionRoundingModeINTEL* with matching _Target Width_ operand, the rounding mode specified by *FunctionRoundingModeINTEL* is applied and *RoundingModeRTNINTEL* is ignored. +
+Only affects instructions operating on a floating-point type whose component width is _Target Width_.
+
+_Target Width_ is an unsigned 32-bit integer.
+| <<Literal, 'Literal'>> _Target Width_ | *RoundToInfinityINTEL*
+| 5622 | *FloatingPointModeALTINTEL* +
+The default floating-point operation mode for floating-point arithmetic and conversions instructions must be the Alternative Floating-Point Mode as specified by the client API.
+If an instruction defines a floating-point operation mode in its description, that floating-point operation mode is applied and *FloatingPointModeALTINTEL* is ignored.
+If an instruction does not define a floating-point operation mode in its description and is inside of a function decorated with *FunctionFloatControlINTEL* with matching _Target Width_ operand,
+the floating-point operation mode specified by *FunctionFloatControlINTEL* is applied and *FloatingPointModeALTINTEL* is ignored. +
+Only affects instructions operating on a floating-point type whose component width is _Target Width_.
+
+_Target Width_ is an unsigned 32-bit integer.
+| <<Literal, 'Literal'>> _Target Width_ | *FloatingPointModeINTEL*
+| 5623 | *FloatingPointModeIEEEINTEL* +
+The default floating-point operation mode for floating-point arithmetic and conversions instructions must be the IEEE 754 Mode as specified by the client API.
+If an instruction defines a floating-point operation mode in its description, that floating-point operation mode is applied and *FloatingPointModeIEEEINTEL* is ignored.
+If an instruction does not define a floating-point operation mode in its description and is inside of a function decorated with *FunctionFloatControlINTEL* with matching _Target Width_ operand,
+the floating-point operation mode specified by *FunctionFloatControlINTEL* is applied and *FloatingPointModeIEEEINTEL* is ignored. +
+Only affects instructions operating on a floating-point type whose component width is _Target Width_.
+
+_Target Width_ is an unsigned 32-bit integer.
+| <<Literal, 'Literal'>> _Target Width_ | *FloatingPointModeINTEL*
+|====
+--
+Modify Section 3 Binary form, add new sub-sections after 3.16 FP Rounding Mode:
+--
+[[FP_Denorm_Mode]]*3.17. FP Denorm Mode*
+
+Denormalized values handling mode.
+
+[cols="^.^1,15,15",options="header",width = "50%"]
+|====
+2+^.^| FP Denorm Mode | Enabling Capabilities
+| 0 | *Preserve* +
+Denormalized values must be preserved. |
+| 1 | *FlushToZero* +
+Denormalized values must be flushed to zero. |
```

Comment:
We already have corresponding execution modes whose description uses "is flushed to zero" or "is preserved". We could do the same here (instead of _must_ ) and defer to Client API spec.
Note: Vulkan spec defines a list of instructions that _must_ flush to zero or preserve. Other instructions _may_ flush to zero or preserve.